pineappl_capi/src/lib.rs

```diff
@@ -1357,3 +1355,253 @@ pub unsafe extern "C" fn pineappl_string_delete(string: *mut c_char) {
+    interpolations: *const InterpTuples, // takes: `struct InterpTuples interpdata[] = { {...} }`
+) -> Box<Grid> {
+    // PID Basis
+    let pid_basis = match unsafe {
+        CStr::from_ptr(pid_basis)
+            .to_str()
+            .expect("Failed to convert `pid_basis` into a proper string.")
+    } {
+        "Evol" => PidBasis::Evol,
+        "Pdg" => PidBasis::Pdg,
+        _ => todo!(),
+    };
+
+    // Luminosity channels
+    let channels = unsafe { &*channels };
+
+    // Perturbative orders
+    let order_params = unsafe { slice::from_raw_parts(order_params, 4 * orders) };
+    let orders: Vec<_> = order_params
+        .chunks(4)
+        .map(|s| Order {
+            alphas: s[0].try_into().unwrap(),
+            alpha: s[1].try_into().unwrap(),
+            logxir: s[2].try_into().unwrap(),
+            logxif: s[3].try_into().unwrap(),
+            logxia: s[4].try_into().unwrap(),
+        })
+        .collect();
+
+    // Bin limits
+    let bin_limits = unsafe { slice::from_raw_parts(bin_limits, bins + 1).to_vec() };
+
+    // Types of convolutions
+    let mut convolutions = Vec::new();
+    for i in 0..nb_convolutions {
+        let c_conv = unsafe {
+            let str_ptr = *convolution_types.add(i);
+            CStr::from_ptr(str_ptr)
+                .to_str()
+                .expect("Failed to convert one of the `convolution_types` into a proper string.")
+        };
+
+        let pid_value = unsafe { *pdg_ids.add(i) };
+
+        let cvtype = match c_conv {
+            "UnpolPDF" => Conv::new(ConvType::UnpolPDF, pid_value),
+            "PolPDF" => Conv::new(ConvType::PolPDF, pid_value),
+            "UnpolFF" => Conv::new(ConvType::UnpolFF, pid_value),
+            "PolFF" => Conv::new(ConvType::PolFF, pid_value),
+            _ => todo!(),
+        };
+        convolutions.push(cvtype);
+    }
+
+    // Grid interpolations
+    let mut interp_vecs = Vec::new();
+    let interp_slices = unsafe { std::slice::from_raw_parts(interpolations, nb_convolutions + 1) };
+    for interp_ref in interp_slices {
+        interp_vecs.push(construct_interpolation(interp_ref));
+    }
+
+    // Kinematics
+    let mut kinematics_vec = Vec::new();
+    for k in 0..=nb_convolutions {
+        let kin = unsafe {
+            let kin_ptr = *kinematics.add(k);
+            CStr::from_ptr(kin_ptr)
+                .to_str()
+                .expect("Failed to convert one of the `kinematics` types into a proper string.")
+        };
+
+        let kin_type = match kin {
+            "scale" => Kinematics::Scale(0),
+            "x1" => Kinematics::X1,
+            "x2" => Kinematics::X2,
+            _ => todo!(),
+        };
+        kinematics_vec.push(kin_type);
+    }
+
+    Box::new(Grid::new(
+        pid_basis,
+        channels.0.clone(),
+        orders,
+        bin_limits,
+        convolutions,
+        interp_vecs,
+        kinematics_vec,
+        Scales {
```

Comment:
This the user should also be able to set, because we might have completely different values for renormalization, factorization and fragmentation scales.

Reply:
Addressed in [4d0f526](https://github.com/NNPDF/pineappl/pull/299/commits/4d0f526634fa0c2e40966ba30612d15dcf4fed15). The other comments I marked resolved instead have been addressed in [93d4d62](https://github.com/NNPDF/pineappl/pull/299/commits/93d4d625decc4206b7073397797a4c3ac2f20253) but please check (I mainly resolved them in order to know myself which one I've addressed xp ). I also added another "temporary" example to check this.

The remaining comments I still have to address.